index.bs

```diff
@@ -504,6 +505,67 @@ QuicTransportBase includes DatagramTransport;
 QuicTransportBase includes WebTransport;
 </pre>
 
+## Methods ##  {#quic-transport-base-methods}
+
+: <dfn for="QuicTransportBase" method>getStats()</dfn>
+:: Gathers stats for this <code><a>QuicTransportBase</a></code>'s QUIC
+   connection and reports the result asynchronously.</p>
+
+   When close is called, the user agent MUST run the following steps:
+     1. Let |transport| be the QuicTransportBase on which `getStats` is invoked.
+     1. If |transport|'s {{[[WebTransportState]]}} is `"new"`, then abort
+        these steps.
+     1. Let |p| be a new promise.
+     1. Return |p| and continue the following steps in background.
+         1. Gather the stats from the underlying QUIC connection.
+         1. Once stats have been gathered, resolve |p| with the
+            {{QuicTransportStats}} object, representing the gathered stats.
+
+## `QuicTransportStats` Dictionary ##  {#quic-transport-stats}
+
+The <dfn dictionary>QuicTransportStats</dfn> dictionary includes information
+on QUIC connection level stats.
+
+<pre class="idl">
+dictionary QuicTransportStats {
+  DOMHighResTimeStamp timestamp;
+  unsigned long long bytesSent;
+  unsigned long long packetsSent;
+  unsigned long numOutgoingStreamsCreated;
+  unsigned long numIncomingStreamsCreated;
+  unsigned long long bytesReceived;
+  unsigned long long packetsReceived;
+  DOMHighResTimeStamp minRttUs;
+  unsigned long numReceivedDatagramsDropped;
+};
+</pre>
+
+The dictionary SHALL have the following attributes:
+
+: <dfn for="QuicTransportStats" dict-member>timestamp</dfn>
+:: The `timestamp` for when the stats are gathered, relative to the
+UNIX epoch (Jan 1, 1970, UTC).
+: <dfn for="QuicTransportStats" dict-member>bytesSent</dfn>
+:: The amount of bytes sent on the QUIC connection, including retransmissions.
+Does not include UDP or any other outer framing.
+: <dfn for="QuicTransportStats" dict-member>packetsSent</dfn>
+:: The amount of packets sent on the QUIC connection, including retransmissions.
+: <dfn for="QuicTransportStats" dict-member>numOutgoingStreamsCreated</dfn>
+:: The amount of outgoing QUIC streams created on the QUIC connection.
+: <dfn for="QuicTransportStats" dict-member>numIncomingStreamsCreated</dfn>
+:: The amount of incoming QUIC streams created on the QUIC connection.
+: <dfn for="QuicTransportStats" dict-member>bytesReceived</dfn>
+:: The amount of total bytes received on the QUIC connection, including
+duplicate data for streams. Does not include UDP or any other outer framing.
+: <dfn for="QuicTransportStats" dict-member>packetsReceived</dfn>
+:: The amount of total packets received on the QUIC connection, including
+packets that were not processable.
+: <dfn for="QuicTransportStats" dict-member>minRttUs</dfn>
+:: The minimum RTT.
```

Comment:
Over what time window?  The entire connection?

Reply:
The entire connection. It looks like QUICHE does this for the entire time the connection has been open:
https://cs.chromium.org/chromium/src/net/third_party/quiche/src/quic/core/congestion_control/rtt_stats.cc?dr=CSs&g=0&l=43

Reply:
I updated to "The minimum RTT observed on the entire connection." Leaving this comment open and leaving it to you to resolve or suggest something different. I think this is the most straight forward thing to do, but maybe in the future  a given time window would be better.